cellfinder_napari/train/train.py

```diff
@@ -0,0 +1,183 @@
+from lib2to3.pgen2.token import OP
+from pathlib import Path
+from typing import Optional
+
+from cellfinder_core.download.models import model_weight_urls
+from cellfinder_core.train.train_yml import models
+from magicgui import magicgui
+from magicgui.widgets import FunctionGui
+from napari.qt.threading import thread_worker
+
+from cellfinder_napari.utils import (
+    header_label_widget,
+    html_label_widget,
+    widget_header,
+)
+
+from .train_containers import (
+    MiscTrainingInputs,
+    OptionalNetworkInputs,
+    OptionalTrainingInputs,
+    TrainingDataInputs,
+)
+
+
+def train() -> FunctionGui:
+    @magicgui(
+        header=header_label_widget,
+        training_label=html_label_widget("Network training", "h3"),
+        **TrainingDataInputs.widget_representation(),
+        **OptionalNetworkInputs.widget_representation(),
+        **OptionalTrainingInputs.widget_representation(),
+        **MiscTrainingInputs.widget_representation(),
+        call_button=True,
+        reset_button=dict(widget_type="PushButton", text="Reset defaults"),
+    )
+    def widget(
+        header,
+        training_label,
+        data_options: dict,
+        yaml_files: Path,
+        output_directory: Path,
```

Comment:
Thoughts on this @alessandrofelder ?

Reply:
Napari crashes ~silently when I change this to `os.PathLike` so I will avoid it for now at least.
This is presumably because magicgui doesn't know what widget is should create for an `os.PathLike` object, but it does for `Path`?

Reply:
@dstansby I agree when we know that should be strict. I wasn't sure if we knew here or not